contrib/opencensus-ext-stackdriver/opencensus/ext/stackdriver/stats_exporter/__init__.py

```diff
@@ -131,209 +130,174 @@ def options(self):
+            # this just modifies points, no return
+            self._convert_point(metric, ts, point, sd_point)
+        return series
+
+    def _convert_point(self, metric, ts, point, sd_point):
+        """Convert an OC metric point to a SD point."""
+        if (metric.descriptor.type ==
+                metric_descriptor.MetricDescriptorType.CUMULATIVE_DISTRIBUTION):
+
+            sd_dist_val = sd_point.value.distribution_value
+            sd_dist_val.count = point.value.count
+            sd_dist_val.sum_of_squared_deviation = point.value.sum_of_squared_deviation
+
+            assert sd_dist_val.bucket_options.explicit_buckets.bounds == []
+            sd_dist_val.bucket_options.explicit_buckets.bounds.extend(
+                [0.0] +
+                list(map(float, point.value.bucket_options.type_.bounds))
+            )
+
+            assert sd_dist_val.bucket_counts == []
+            sd_dist_val.bucket_counts.extend(
+                [0] +
+                [bb.count for bb in point.value.buckets]
+            )
+
+        elif (metric.descriptor.type ==
+              metric_descriptor.MetricDescriptorType.CUMULATIVE_INT64):
+            sd_point.value.int64_value = int(point.value.value)
+
+        elif (metric.descriptor.type ==
+              metric_descriptor.MetricDescriptorType.CUMULATIVE_DOUBLE):
+            sd_point.value.double_value = float(point.value.value)
+
+        elif (metric.descriptor.type ==
+              metric_descriptor.MetricDescriptorType.GAUGE_INT64):
+            sd_point.value.int64_value = int(point.value.value)
+
+        elif (metric.descriptor.type ==
+              metric_descriptor.MetricDescriptorType.GAUGE_DOUBLE):
+            sd_point.value.double_value = float(point.value.value)
+
         else:
-            raise Exception(
-                "unsupported aggregation type: %s" % type(view_aggregation))
+            md_type_name = metric_descriptor.MetricDescriptorType.to_name(
+                metric.descriptor.type)
+            raise TypeError("Unsupported metric type: {}".format(md_type_name))
+
+        start = datetime.strptime(ts.start_timestamp, EPOCH_PATTERN)
+        end = point.timestamp
+
+        timestamp_start = (start - EPOCH_DATETIME).total_seconds()
+        timestamp_end = (end - EPOCH_DATETIME).total_seconds()
+
+        sd_point.interval.end_time.seconds = int(timestamp_end)
+
+        secs = sd_point.interval.end_time.seconds
+        sd_point.interval.end_time.nanos = int((timestamp_end - secs) * 1e9)
+
+        start_time = sd_point.interval.start_time
+        start_time.seconds = int(timestamp_start)
+        start_secs = start_time.seconds
+        start_time.nanos = int((timestamp_start - start_secs) * 1e9)
+
+    def get_descriptor_type(self, oc_md):
+        """Get a SD descriptor type for an OC metric descriptor."""
+        return namespaced_view_name(oc_md.name, self.options.metric_prefix)
+
+    def get_metric_descriptor(self, oc_md):
```

Comment:
Minor suggestion, probably we can use a predefined dict to do the lookup?

Reply:
agreed. In fact one already exists for gauge: `metric_utils.is_gauge`. I think it would make sense to create one for cumulative and for the value types as well.

Reply:
Done in bce0b86, good call.